aikau/src/test/resources/alfresco/forms/controls/BaseFormTest.js

```diff
@@ -29,146 +29,193 @@ define(["intern!object",
+               .then(function(payload) {
+                  assert.propertyVal(payload, "control", "wibble", "Did not autosave updated value");
+                  assert.propertyVal(payload, "alfValidForm", true, "Did not autosave validity property");
+               });
+         },
+
+         "Updating to invalid value does not autosave form": function() {
+            return browser.findById("CLEAR_AUTOSAVE_1")
+               .click()
+               .clearLog()
+               .getAllPublishes("AUTOSAVE_FORM_1")
+               .then(function(payloads) {
+                  assert.lengthOf(payloads, 0, "Published form when invalid");
+               });
+         },
+
+         "Autosave on invalid flag publishes invalid form": function() {
+            return browser.findByCssSelector("#AUTOSAVE_INVALID_FORM_FIELD .dijitInputInner")
+               .clearLog()
+               .clearValue()
+               .pressKeys(keys.BACKSPACE) // Need to trigger an update!
+               .getLastPublish("AUTOSAVE_FORM_2")
+               .then(function(payload) {
+                  assert.propertyVal(payload, "control", "", "Did not autosave updated, invalid value");
+                  assert.propertyVal(payload, "alfValidForm", false, "Did not autosave validity property");
+               });
+         },
+
+         "Autosaving with defined payload mixes payload into form values": function(){
+            return browser.findByCssSelector("#AUTOSAVE_FORM_FIELD") // Need to get session to check for publish
+               .getLastPublish("AUTOSAVE_FORM_2")
+               .then(function(payload) {
+                  assert.propertyVal(payload, "customProperty", "awooga", "Did not mix custom payload into form values");
+               });
+         },
+
+         "Post Coverage Results": function() {
+            TestCommon.alfPostCoverageResults(this, browser);
+         }
+      };
+   });
+
+   // This test reloads the page to clear any previous focus
+   registerSuite(function(){
+      var browser;
+
+      return {
+         name: "Base Form Control Tests - Focus and Enter to Submit",
+
+         setup: function() {
+            browser = this.remote;
+            return TestCommon.loadTestWebScript(this.remote, "/BaseForm", "Base Form Control Tests").end();
+         },
+
+         beforeEach: function() {
+            browser.end();
+         },
+
+         "Check the form fields are correctly focused": function() {
+            return browser.findByCssSelector("#FOCUS_AND_SUBMIT_ON_ENTER_FORM #FORM_FIELD1 div.dijitFocused")
+               .end()
+
+            .findByCssSelector("#FOCUS_AND_SUBMIT_ON_ENTER_FORM #FORM_FIELD2 div.dijitFocused")
+               .then(function () {
+                  assert.fail(null, null, "Form field 2 should not be focused");
+               },
+               function () {
+                  
+               });
+         },
+
+         "Check the form submits by the enter key": function() {
```

Comment:
There is no test that a form is NOT submitted when the new configuration value isn't applied.